From 193d9c51e519ddf5652ffa9be94dfaa258205ea3 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:16 -0400
Subject: [ethernet] qlcnic: Fix function return error check

Message-id: <1408626422-3941-93-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88618
O-Subject: [PATCH RHEL 7.1 092/138] qlcnic: Fix function return error check
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shahed Shaikh <shahed.shaikh@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id fc49beaee2a410402f49fd21c81a37a863b7a9ba

Driver was treating -ve return value as success in case of
qlcnic_enable_msi_legacy() failure

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index a6b9a7f..d2e9ea8 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -824,7 +824,7 @@ static int qlcnic_82xx_setup_intr(struct qlcnic_adapter *adapter)
    qlcnic_disable_multi_tx(adapter);
 
    err = qlcnic_enable_msi_legacy(adapter);
-   if (!err)
+   if (err)
     return err;
   }
  }
-- 
1.7.1